src/components/OrdersWidget/index.tsx

```diff
@@ -160,7 +164,16 @@ const OrdersForm = styled.div`
 `
 
 const OrdersWidget: React.FC = () => {
-  const orders = useOrders()
+  const allOrders = useOrders()
+
+  const [orders, setOrders] = useSafeState<AuctionElement[]>(allOrders)
+
+  useEffect(() => {
+    const filteredOrders = allOrders.filter(order => order.validUntil > dateToBatchId())
```

Comment:
Why do you filter in `useEffect`? I don't see why it needs to be done async

Reply:
my only comment, otherwise looks good

Reply:
Because `useOrders` is async. At first it returns an empty list https://github.com/gnosis/dex-react/blob/develop/src/hooks/useOrders.ts#L7

Reply:
And I'm using state because once delete is complete I want to remove the orders from the UI right away.